analytics.py

```diff
@@ -0,0 +1,213 @@
+from .utils import *
+
+def find_largest_city(gj):
+  """
+  Iterate through a geojson feature collection and
+  find the largest city.  Assume that the key
+  to access the maximum population is 'pop_max'.
+
+  Parameters
+  ----------
+  gj : dict
+       A GeoJSON file read in as a Python dictionary
+
+  Returns
+  -------
+  city : str
+         The largest city
+
+  population : int
+               The population of the largest city
+  """
+  city = None
+  max_population = 0
+  for n in gj["features"]:
+      properties = n["properties"]
+      if (properties["pop_max"] > max_population):
+          max_population = properties["pop_max"]
+          city = properties["adm1name"]
+
+  return city, max_population
+
+
+def alaska_points(gj):
+  # Find coordinates from Alaska
+  alaska_points = []
+  for n in gj["features"]:
+      properties = n["properties"]
+      state_name = properties["adm1name"]
+      if state_name == "Alaska":
+          alaska_points.append(n)
+      else:
+          continue
+
+  return alaska_points
+
+
+def mean_center(points):
+  """
+  Given a set of points, compute the mean center
+
+  Parameters
+  ----------
+  points : list
+       A list of points in the form (x,y)
+
+  Returns
+  -------
+  x : float
+      Mean x coordinate
+
+  y : float
+      Mean y coordinate
+  """
+  x = 0
+  y = 0
+  number_of_points = 0
+
+  for p in points: 
+      x += p[0]
+      y += p[1]
+      number_of_points += 1
+
+  x = x / number_of_points
+  y = y / number_of_points
+
+  return x, y
+
+
+def average_nearest_neighbor_distance(points):
+  """
+  Given a set of points, compute the average nearest neighbor.
+
+  Parameters
+  ----------
+  points : list
+           A list of points in the form (x,y)
+
+  Returns
+  -------
+  mean_d : float
+           Average nearest neighbor distance
+
+  References
+  ----------
+  Clark and Evan (1954 Distance to Nearest Neighbor as a
+   Measure of Spatial Relationships in Populations. Ecology. 35(4)
+   p. 445-453.
+  """
+  mean_d = 0
+  temp_p = None
+  last_p = None
+
+  for p in points:
+	  for neighbor in p:
```

Comment:
This is saying
```python
for every point:
    for each coordinate in the point:
        # do stuff
```
Is that what you want or do you want to loop over the list twice?